pkg/distributor/push.go

```diff
@@ -80,22 +82,23 @@ func handler(
 	allowSkipLabelNameValidation bool,
 	limits *validation.Overrides,
 	push PushFunc,
+	logger log.Logger,
 	parser parserFunc,
 ) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		ctx := r.Context()
-		logger := log.WithContext(ctx, log.Logger)
+		logger := utillog.WithContext(ctx, logger)
```

Comment:
```suggestion
		logger = utillog.WithContext(ctx, logger)
```
we want to modify the origin logger instead of creating new ones, so it could be propagated.

Reply:
Which  _semantic difference_ does your suggestion make? The following change won't modify the original logger (we don't want to modify the original logger in any case):

```golang
logger = utillog.WithContext(ctx, logger)
```